src/uploadx/lib/retry-handler.ts

```diff
@@ -50,7 +57,7 @@ export class RetryHandler {
     if (this.config.shouldRestartCodes.indexOf(code) !== -1) {
       return ErrorType.NotFound;
     }
-    if (code < 400 || code >= 500 || this.config.shouldRetryCodes.indexOf(code) !== -1) {
+    if (this.config.shouldRetry(code, this.attempts)) {
```

Comment:
`this.attempts` is not in the signature

Reply:
is not used in `defaultRetryConfig`, but `RetryConfig.shouldRetry: (code: number, attempts: number) => boolean`